### ISSUE TITLE
example for link prediction

### ISSUE THREAD

**Issue by jwwu666** [dmlc/dgl#2353]
## ❓ Questions and Help
In your provided examples, the GraphSAGE for unsupervised learning is not done link prediction on homogeneous graphs as you say.

**Comment by VoVAllen** [comment]
Which example are you referring to?

**Comment by jwwu666** [comment]
dgl/examples/pytorch/graphsage/train_sampling_unsupervised.py

**Comment by VoVAllen** [comment]
What's the problem you met? Or what feature are you looking for?

**Comment by jwwu666** [comment]
The code in this file does not seem to perform the link prediction task as you said. What task is this code doing?

**Comment by Veronicium** [comment]
Same. It seems that it's doing node classification. The "compute_acc" function takes in ground truth node labels and outputs F1 score

**Comment by VoVAllen** [comment]
Why do you think it's not link prediction? It samples from big graph and generates positive and negative edges and calculates the loss at https://github.com/dmlc/dgl/blob/master/examples/pytorch/graphsage/train_sampling_unsupervised.py#L120-L133

**Comment by jwwu666** [comment]
The inference function of the model is

**Comment by jwwu666** [comment]
The inference function seems to be doing node classification. Although the training period is surely training a link predictor.

**Comment by yzh119** [comment]
@jwwu666 Yes you are right, the code did node classification instead of link prediction.
In training phase, the target is to predict each node's neighbor (encourage the score in positive graph to be higher than that in negative graph), which somewhat looks like link prediction but it's not.
In inference phase, the target is node classification.
